### PR TITLE
KAFKA-7801: TopicCommand should not be able to alter transaction topic partition count

### PR DESCRIPTION
**Pull request by huxihx**
To keep align with the way it handles the offset topic, TopicCommand should not be able to alter transaction topic partition count.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by huxihx** [comment]
retest this please

**Comment by huxihx** [comment]
@ijuma Please review this minor change. Thanks.

**Review by viktorsomogyi — COMMENTED** [review]
I think ideally we want to restrict this with ACLs, however #5683 will allow that functionality, so I guess it's OK to do this here.

**Comment by huxihx** [comment]
retest this please

**Comment by ijuma** [comment]
Thanks for the PR. Can we add a test?

File: core/src/main/scala/kafka/admin/TopicCommand.scala [hunk 1]
```diff
@@ -318,8 +318,8 @@ object TopicCommand extends Logging {
         }
 
         if(tp.hasPartitions) {
-          if (topic == Topic.GROUP_METADATA_TOPIC_NAME) {
-            throw new IllegalArgumentException("The number of partitions for the offsets topic cannot be changed.")
+          if (Topic.INTERNAL_TOPICS.contains(topic)) {
+            throw new IllegalArgumentException("The number of partitions for the offsets topic or the transaction topic cannot be changed.")
```
Comment (omkreddy) [inline core/src/main/scala/kafka/admin/TopicCommand.scala]: Can we use comma separated string of Topic.INTERNAL_TOPICS  in the error message?

**Review by omkreddy — APPROVED** [review]
@huxihx Thanks for the PR. LGTM.

**Pull request closed without merging**